### ISSUE TITLE
Temporal probe reliability and safety guardrails

### ISSUE THREAD

**Issue by justinshenk** [justinshenk/temporal-awareness#17]
## Motivation

Temporal probes detect internal planning horizons, but for deployment in safety-critical monitoring, we need to understand their reliability characteristics and failure modes. This experiment applies production safety engineering principles to temporal probe deployment.

## Research Questions

1. What are the reliability characteristics of temporal probes under distribution shift?
2. Can we build confidence-calibrated probes that know when they don't know?
3. What safety guardrails are needed for production temporal monitoring?
4. How do probe failure modes correlate with model failure modes?

### Phase 1: Reliability benchmarking

Systematic evaluation of temporal probes:

```python
# Evaluate probe reliability across conditions
conditions = {
    'in_distribution': standard_test_set,
    'domain_shift': out_of_domain_prompts,
    'adversarial': red_team_inputs,  # from #14
    'edge_cases': ambiguous_temporal_content,
    'distribution_shift': different_writing_styles
}

metrics = {
    'accuracy': classification_accuracy,
    'calibration': expected_calibration_error,
    'confidence_reliability': confidence_vs_correctness,
    'failure_detection': knows_when_wrong
}
```

### Phase 2: Confidence calibration

Build probes that output calibrated confidence scores:

```python
# Train probe with temperature scaling / Platt scaling
# Goal: P(correct | confidence=0.8) ≈ 0.8

def calibrated_temporal_probe(activations):
    raw_logits = probe(activations)
    calibrated_probs = temperature_scale(raw_logits)
    
    return {
        'prediction': argmax(calibrated_probs),
        'confidence': max(calibrated_probs),
        'uncertainty': entropy(calibrated_probs),
        'abstain': confidence < threshold
    }
```

### Phase 3: Safety guardrails framework

Design guardrails for temporal monitoring deployment:

1. **Abstention mechanism**: Probe refuses to classify when uncertain
2. **Anomaly detection**: Flag inputs that look different from training
3. **Consistency checks**: Cross-validate with multiple probe heads
4. **Fallback hierarchy**: What happens when probe fails?

```python
class TemporalMonitorWithGuardrails:
    def __init__(self, probe, confidence_threshold=0.7):
        self.probe = probe
        self.threshold = confidence_threshold
        self.anomaly_detector = train_anomaly_detector()
    
    def monitor(self, activations):
        # Check for out-of-distribution
        if self.anomaly_detector.is_anomalous(activations):
            return {'status': 'ABSTAIN', 'reason': 'OOD input'}
        
        pred, conf = self.probe(activations)
        
        if conf < self.threshold:
            return {'status': 'LOW_CONFIDENCE', 'prediction': pred}
        
        return {'status': 'OK', 'prediction': pred, 'confidence': conf}
```

### Phase 4: Failure mode analysis

Systematically characterize when and why probes fail:

- Failure taxonomy (false positives vs false negatives)
- Correlation with input characteristics
- Correlation with model internal states
- Early warning signals for probe degradation

## Connection to Other Issues

- Uses adversarial inputs from #14 (Red-teaming)
- Informs deployment of probes from Track A
- Complements #5 (Eval awareness) with reliability analysis
- Production considerations for #16 (Multi-agent monitoring)

## Success Criteria

| Metric | Target |
|--------|--------|
| Calibration error (ECE) | < 0.10 |
| OOD detection rate | > 90% |
| Appropriate abstention rate | 5-15% |
| Failure prediction accuracy | > 70% |

## Deliverables

1. `scripts/experiments/probe_reliability.py` - Benchmarking framework
2. `src/guardrails/` - Safety guardrails module
3. Reliability report with failure taxonomy
4. Recommendations for production deployment

## Track

Track B (Experiments - parallelizable)

## Thrust

Robustness & Failure Modes
